src/Tmds.Ssh/Curve25519KeyExchange.cs

```diff
@@ -0,0 +1,135 @@
+// This file is part of Tmds.Ssh which is released under MIT.
+// See file LICENSE for full license details.
+
+using System.Buffers;
+using System.Security.Cryptography;
+using System.Numerics;
+using Microsoft.Extensions.Logging;
+using Org.BouncyCastle.Crypto.Generators;
+using Org.BouncyCastle.Crypto.Parameters;
+using Org.BouncyCastle.Security;
+using Org.BouncyCastle.Crypto.Prng;
+using Org.BouncyCastle.Crypto;
+using Org.BouncyCastle.Crypto.Agreement;
+
+namespace Tmds.Ssh;
+
+// Curve25519 Key Exchange: https://datatracker.ietf.org/doc/html/rfc8731
+class Curve25519KeyExchange : KeyExchange
+{
+    private readonly HashAlgorithmName _hashAlgorithmName = HashAlgorithmName.SHA256;
+
+    public override async Task<KeyExchangeOutput> TryExchangeAsync(KeyExchangeContext context, IHostKeyVerification hostKeyVerification, Packet firstPacket, KeyExchangeInput input, ILogger logger, CancellationToken ct)
+    {
+        var sequencePool = context.SequencePool;
+        var connectionInfo = input.ConnectionInfo;
+
+        AsymmetricCipherKeyPair x25519KeyPair;
+        using (var _randomGenerator = new CryptoApiRandomGenerator())
+        {
+            var x25519KeyPairGenerator = new X25519KeyPairGenerator();
+            x25519KeyPairGenerator.Init(new X25519KeyGenerationParameters(new SecureRandom(_randomGenerator)));
+            x25519KeyPair = x25519KeyPairGenerator.GenerateKeyPair();
+        }
+
+        // Send ECDH_INIT.
+        var q_c = ((X25519PublicKeyParameters)x25519KeyPair.Public).GetEncoded();
```

Comment:
`var` -> `byte[]`

The code style only uses `var` when the right hand side is a constructor or explicitly contains the returned type as part of the member name being called.